Java/core/src/main/java/com/amazon/randomcutforest/sampler/CompactSampler.java

```diff
@@ -372,8 +384,21 @@ protected float computeWeight(long sequenceIndex) {
         while (randomNumber == 0d) {
             randomNumber = random.nextDouble();
         }
+        mostRecentTimeStamp = (mostRecentTimeStamp < sequenceIndex) ? sequenceIndex : mostRecentTimeStamp;
```

Comment:
question. when does mostrecentts increase to be greater than sequence index?

Reply:
mostRecentTimestamp corresponds to the maximum sequence index seen. The hidden assumption is that the sequence-indices and timestamps where lambda is changed are monotonically (non-decreasing) ordered. The sequence indices can be out-of-order among themselves. The assumption is that once lambda is updated, we just need to know the maximum sequence index seen so far. But a good callout - will add the comment.

Reply:
To put it another way, we want to be able to support a batch training mode where we don't want a strict requirement that points are submitted in timestamp order. However, for the computation for the accumulated lambda value used in the sampling weight to be correct, we need the following:
* it has to be computed using the maximum seen timestamp (not necessarily the most recently submitted timestamp)
* after the lambda value changes (i.e., after a call to `setLambda`, we can no longer accept points with a timestamp smaller than the current value of `mostRecentTimestamp`

With that in mind, I think we should add a validation to `acceptSample` to check that submitted timestamps are > `lastUpdateOfLambda`.

Reply:
Done!

Reply:
is the difference of timestamp and sequenceindex necessary? or can they be named consistently? like sequenceIndex and largestSequenceIndex?

Reply:
largestSequenceIndexSeen makes sense. Will change.